onnxscript/function_libs/torch_aten/ops/core.py

```diff
@@ -5756,10 +5756,44 @@ def aten_type_as(self: TensorType, other: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-def aten_unfold(self: TensorType, dimension: int, size: int, step: int) -> TensorType:
+@torch_op("aten::unfold", trace_only=True)  # FIXME: Seems ast.For was not supported
+def aten_unfold(self: TTensor, dimension: int, size: int, step: int) -> TTensor:
     """unfold(Tensor(a) self, int dimension, int size, int step) -> Tensor(a)"""
 
-    raise NotImplementedError()
+    self_shape = op.Shape(self)
+    self_rank = op.Size(self_shape)
+    if self_rank == 0:
+        result = op.Unsqueeze(self, 0)
+    else:
+        dims = op.Constant(value_ints=[dimension])
+        dim_size = op.Gather(self_shape, dims, axis=0)
+        # target = ((i-size)/step + 1) * step != i-size+step
+        target_end = op.Squeeze(((dim_size - size) / step + 1) * step)
+        seq_result = op.SequenceEmpty()
+
+        for i in range(target_end):
```

Comment:
Using trace only will be tricky because we don’t know what target end is at trace time